src/components/Button.tsx

```diff
@@ -55,15 +57,15 @@ const commonClasses = (
     outlineStyle('focus:outline-none'),
     opacity({ 'opacity-50': !available }),
     boxShadow({
-      'shadow-2xl': available,
-      'hover:shadow-lg': available,
-      'active:shadow-button-active': available,
+      'shadow-2xl': available && notDefaultType,
```

Comment:
You're doing `avaliable && notDefaultType` 3 times, should we extract it into `const`?

Reply:
Done